src/tactic/command/global_search_trigger.py

```diff
@@ -86,6 +104,77 @@ def execute(my):
         data.update( my.cleanup(caller.get_value("description", no_exception=True) ))
         data.update( my.cleanup(caller.get_value("keywords", no_exception=True) ))
 
+
+        # If keywords_data column exists and collection is being changed
+        if has_keywords_data:
+            
+            update_data = input.get("update_data")
+            if "user_keywords" in update_data:
+                has_user_keywords = True
+            else:
+                has_user_keywords = False
+
+            if is_collection:
+                if input.get("mode") == "update" and "name" in update_data:
+                    rename_collection = True
+
+                # New Collection created
+                if input.get("is_insert"):
+                    collection_keywords = update_data.get("user_keywords")
+                    collection_name = update_data.get("name")
+
+                    keywords_data = sobj.get_json_value("keywords_data", {})
+                    keywords_data['user'] = "%s %s" % (collection_name, collection_keywords)
+
+                    sobj.set_json_value("keywords_data", keywords_data)
+                    sobj.commit(triggers=False)
+                    my.set_searchable_keywords(sobj)  
+
+                # If collection is renamed
+                elif rename_collection:
+                    collection_name = update_data.get("name")
+                    keywords_data = sobj.get_json_value("keywords_data", {})
+
+                    if 'user' in keywords_data:
+                        user = keywords_data.get('user')
+                        old_collection_name = input.get("prev_data").get("name")
+
+                        user = user.replace(old_collection_name, collection_name)
+                        keywords_data['user'] = user
+                        
+                        sobj.set_json_value("keywords_data", keywords_data)
+                        sobj.commit(triggers=False)
+
+                        my.update_user_keywords(sobj, user, base_search_type)
+                        my.set_searchable_keywords(sobj)
+
+                # If user defined keywords column is changed 
+                elif has_user_keywords:
+                    user_keywords = input.get("update_data").get("user_keywords")
+
+                    if not user_keywords:
+                        user_keywords = ""
```

Comment:
why is these 2 lines if not user_keywords, user_keywords = '' unique to collection?